x/intrarelayer/types/proposal.go

```diff
@@ -107,6 +107,18 @@ func validateIBC(metadata banktypes.Metadata) error {
 	return nil
 }
 
+// ValidateIntrarelayerDenom checks if a denom is a valid intrarelayer/
+// denomination
+func ValidateIntrarelayerDenom(denom string) error {
+	denomSplit := strings.SplitN(denom, "/", 2)
+
+	if len(denomSplit) != 2 || denomSplit[0] != ModuleName {
+		return fmt.Errorf("invalid denom. %s denomination should be prefixed with the format 'intrarelayer/", denom)
+	}
+
+	return nil
```

Comment:
It should check that the second part is a valid hex address
```suggestion
	return ethermint.ValidateAddress(denomSplit[1]
```

Reply:
Ah nice! I wasn't aware that the address follows.